cmd/agent/common/autodiscovery.go

```diff
@@ -265,23 +268,59 @@ func waitForConfigsFromAD(ctx context.Context, wildcard bool, checkNames []strin
 		}
 	}
 
+	stopChan := make(chan struct{})
 	// add the scheduler in a goroutine, since it will schedule any "catch-up" immediately,
 	// placing items in configChan
 	go AC.AddScheduler("check-cmd", schedulerFunc(func(configs []integration.Config) {
+		var errors []error
 		for _, cfg := range configs {
+			if instanceFilter != "" {
+				instances, filterErrors := filterInstances(cfg.Instances, instanceFilter)
+				if len(filterErrors) > 0 {
+					errors = append(errors, filterErrors...)
+					continue
+				}
+				if len(instances) == 0 {
```

Comment:
I'm not sure if we want to exclude config with no instances here. 
I'm not familiar enough with autodiscovery but is it possible that we get some config without any instance (such as logs config) on this part of the code ?

Reply:
@Kaderinho thanks for the review.

In this case, note that we are here inside the `instanceFilter != ""` condition.

If there is no instances left after the filter being applied I think there is no need to return the config with no instances since the goal for `waitForConfigsFromAD` is to wait configs that match some conditions (checkNames, discoveryMinInstances, instanceFilter).

Maybe I'm missing something, please let me know :)

Reply:
>  is it possible that we get some config without any instance (such as logs config) on this part of the code ?

AFAIK, `waitForConfigsFromAD` is only used for `agent check <INTEGRATION>` command to wait and retrieve integrations configs/instances that should be run.

Reply:
Okay I get it sorry, I missed the fact that you use instanceFilter = "" in `WaitForAllConfigsFromAD` mb